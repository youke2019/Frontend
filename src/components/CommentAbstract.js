import React from 'react'
import {
  View,
  Text,
  FlatList,
  Image,
  ImageBackground,
  StyleSheet,
  TouchableOpacity
} from 'react-native'
import axios from 'axios'
import { ShadowedTitle } from './ShadowedTitle'
import CommentItem from './CommentItem'

const Tag = (props) => {
  return (
    <ImageBackground
      style={styles.tag_style}
      imageStyle={{ resizeMode: 'stretch' }}
      source={{ uri: 'tag' }}
    >
      <Text> {props.text}</Text>
    </ImageBackground>
  )
}

export class CommentAbstractTitle extends React.Component {
  render () {
    const { tags = ['水课', '给分很高', '老师很吵', 'fnu#&2G'] } = this.props
    return (
      <View style={styles.header_container}>
        <ShadowedTitle text={'评论'} uri={'comment'}/>
        {
          tags.map((tag, index) =>
            <Tag text={tag} key={index} style={styles.tag_text}/>)
        }
      </View>
    )
  }
}

export default class CommentAbstract extends React.Component {
  state = {
    comments: []
  }

  componentWillReceiveProps (nextProps, nextContext) {
    this.getCommentData(nextProps)
  }

  getCommentData = (props) => {
    const course_id = props == null ? this.props.course_id : props.course_id
    axios({
      method: 'get',
      url: baseUrl + '/courses/comments/find',
      params: {
        course_id: course_id,
        user_id: this.props.user_info.id
      }
    }).then(response => {
      this.setState({
        comments: response.data
      })
    }).catch(err => {
      console.log('error:' + err)
    })
  }

  render () {
    const firstComm = this.state.comments.length === 0 ? null : this.state.comments[0]

    return (
      <View style={styles.container}>
        <CommentAbstractTitle/>
        <View style={styles.first_comm}>
          <CommentItem
            comment_info={firstComm}
            refresh={this.getCommentData}
          />
        </View>
        <View style={styles.button_container}>
          <TouchableOpacity
            onPress={this.props.onGotoCommentPage}
            style={styles.button_touchable}
            activeOpacity={0.3}
          >
            <Text
              style={styles.button_text}
            >全部评论</Text>
          </TouchableOpacity>
        </View>
        {/* <ReplyBox />*/}
      </View>
    )
  }
}
const styles = StyleSheet.create({
  first_comm: {
    marginHorizontal: 20
  },
  button_text: {
    color: '#ff812e',
    textAlign: 'center'
  },
  button_touchable: {
    borderRadius: 20,
    paddingHorizontal: 55,
    paddingVertical: 5,
    borderWidth:0.5,
    borderColor:'orange',
    backgroundColor: 'white'
  },
  button_container: {
    marginTop: 10,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  header_container: {
    flexDirection: 'row',
    alignItems: 'center',
    flexWrap: 'wrap'
  },
  tag_style: {
    paddingHorizontal: 5,
    marginHorizontal: 5
  },
  tag_text: {
    textAlign: 'center'
  },
  container: {
    borderTopWidth: 15,
    borderTopColor: 'whitesmoke',
    marginBottom: 10
  }

})