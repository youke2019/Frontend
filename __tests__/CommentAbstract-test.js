import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16/build'
import CommentAbstract from '../src/components/CommentAbstract'

configure({adapter: new Adapter()})

jest.mock("../src/utils/DataRequest.js")

test('renders CommentAbstract component correctly', (done) => {
  const tree = renderer.create(
      <CommentAbstract onGotoCommentPage = {()=>{}}  />
  ).toJSON();
  setTimeout(()=>{
    expect(tree).toMatchSnapshot();
    done();
  },1000)
});

describe("<CommentAbstract/> component",()=>{
  const wrapper = shallow(
    <CommentAbstract onGotoCommentPage = {()=>{}} user_info={{id:"test_id"}} course_id={"test_course_id"} />
  );
  const instance = wrapper.instance();
  console.log(instance)
  it("update when receive props",(done)=>{
    instance.componentWillReceiveProps({
      course_id :"test_course_id"
    },null)
    instance.componentWillReceiveProps({
      course_id :"test_course_id"
    },null)
    instance.getCommentData(null)
    setTimeout(()=>{done();},1000) /*wait for async to end*/
  })
})