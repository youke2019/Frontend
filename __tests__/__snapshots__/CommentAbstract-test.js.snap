// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentAbstract component correctly 1`] = `
<View
  style={
    Object {
      "borderTopColor": "whitesmoke",
      "borderTopWidth": 15,
      "marginBottom": 10,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 20,
          "marginVertical": 5,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "position": "relative",
          }
        }
      >
        <Image
          onLoadEnd={[Function]}
          source={
            Object {
              "uri": "comment",
            }
          }
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
              "width": 20,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        />
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "height": 20,
                "justifyContent": "center",
                "resizeMode": "contain",
                "width": 20,
              }
            }
            testID="RNE__Image__placeholder"
          />
        </View>
      </View>
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "position": "relative",
            }
          }
        >
          <Image
            onLoadEnd={[Function]}
            source={
              Object {
                "uri": "yellow_line",
              }
            }
            style={
              Object {
                "height": 10,
                "left": 8,
                "position": "absolute",
                "top": 12,
                "width": 40,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          />
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "height": 10,
                  "justifyContent": "center",
                  "left": 8,
                  "position": "absolute",
                  "top": 12,
                  "width": 40,
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "lineHeight": 25,
              "marginLeft": 10,
              "maxWidth": 250,
              "textAlign": "left",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          评论
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "height": 100,
              "justifyContent": "center",
              "width": 340,
            }
          }
        >
          <Text
            style={
              Object {
                "alignItems": "center",
                "paddingVertical": 20,
              }
            }
          >
             还没有人评论，快来抢占沙发吧😋
          </Text>
        </View>
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        style={
          Object {
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "borderColor": "lightgrey",
              "borderTopWidth": 0.2,
              "bottom": 0,
              "height": 50,
              "paddingHorizontal": 10,
              "paddingVertical": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <RCTScrollView
            keyboardShouldPersistTaps="always"
            style={
              Object {
                "contentSize": 0,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "whitesmoke",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "orange",
                      "borderRadius": 8,
                      "borderWidth": 0.5,
                      "flex": 8,
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "marginVertical": 10,
                      "paddingHorizontal": 10,
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoFocus={true}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="点击输入回复"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "fontSize": 15,
                        "padding": 2,
                        "textAlign": "left",
                        "width": "100%",
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1.5,
                      "marginVertical": 0,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#FDD326",
                        "borderRadius": 8,
                        "height": "auto",
                        "marginHorizontal": 8,
                        "paddingVertical": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "fontFamily": "字魂70号-灵悦黑体",
                          "fontSize": 15,
                          "lineHeight": 15,
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                    >
                       发送
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </Modal>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "orange",
          "borderRadius": 20,
          "borderWidth": 0.5,
          "opacity": 1,
          "paddingHorizontal": 55,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ff812e",
            "textAlign": "center",
          }
        }
      >
        全部评论
      </Text>
    </View>
  </View>
</View>
`;
